router.go

```diff
@@ -34,11 +36,20 @@ func GenerateConditionalRouteHandler(conditionFn RouteConditionFunc, onTrue TxHa
 
 func NewTxRouter() *TxRouter {
 	return &TxRouter{
+		routes:          make(map[uint32]RouteHandler),
 		deliverTxRoutes: make(map[uint32]RouteHandler),
 		checkTxRoutes:   make(map[uint32]RouteHandler),
 	}
 }
 
+func (r *TxRouter) Handle(txID uint32, handler TxHandler) {
+	if _, ok := r.routes[txID]; ok {
+		panic("handler for transaction already registered")
+	}
+	// TODO: remove the GeneratePassthroughRouteHandler once the deliver/checkTxRoutes are gone
+	r.routes[txID] = GeneratePassthroughRouteHandler(handler)
```

Comment:
Nah, once we decide to clean out the legacy code this can go back to the way it was originally:
```golang
r.routes[txID] = handler // routes: make(map[uint32]TxHandler)
```